lidarr/tag_test.go

```diff
@@ -0,0 +1,279 @@
+	tests := []struct {
+		responseStatus   int
+		name             string
+		expectedPath     string
+		responseBody     string
+		withError        error
+		tag              *starr.Tag
+		expectedResponse *starr.Tag
+	}{
+		{
+			name:           "200",
+			expectedPath:   "/api/v1/tag",
+			responseStatus: 200,
+			tag: &starr.Tag{
+				Label: "flac",
+			},
+			responseBody: "{\"label\": \"flac\",\"id\": 1}",
+			expectedResponse: &starr.Tag{
+				Label: "flac",
+				ID:    1,
+			},
+			withError: nil,
+		},
+		{
+			name:             "404",
+			expectedPath:     "/api/v1/tag",
+			responseStatus:   404,
+			responseBody:     `{"message": "NotFound"}`,
+			withError:        starr.ErrInvalidStatusCode,
+			expectedResponse: nil,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				assert.Equal(t, test.expectedPath, r.URL.String())
+				w.WriteHeader(test.responseStatus)
+				_, err := w.Write([]byte(test.responseBody))
+				assert.NoError(t, err)
+			}))
+			client := lidarr.New(starr.New("mockAPIkey", mockServer.URL, 0))
+			output, err := client.AddTag(test.tag)
+			assert.ErrorIs(t, err, test.withError, "error is not the same as expected")
+			assert.EqualValues(t, output, test.expectedResponse, "response is not the same as expected")
+		})
+	}
+}
+
+func TestUpdateTag(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		responseStatus   int
+		name             string
+		expectedPath     string
+		responseBody     string
+		withError        error
+		tag              *starr.Tag
+		expectedResponse *starr.Tag
+	}{
+		{
+			name:           "200",
+			expectedPath:   "/api/v1/tag/1",
+			responseStatus: 200,
+			tag: &starr.Tag{
+				Label: "flac",
+				ID:    1,
+			},
+			responseBody: "{\"label\": \"flac\",\"id\": 1}",
+			expectedResponse: &starr.Tag{
+				Label: "flac",
+				ID:    1,
+			},
+			withError: nil,
+		},
+		{
+			name:         "404",
+			expectedPath: "/api/v1/tag/1",
+			tag: &starr.Tag{
+				Label: "flac",
+				ID:    1,
+			},
+			responseStatus:   404,
+			responseBody:     `{"message": "NotFound"}`,
+			withError:        starr.ErrInvalidStatusCode,
+			expectedResponse: nil,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				assert.Equal(t, test.expectedPath, r.URL.String())
+				w.WriteHeader(test.responseStatus)
+				_, err := w.Write([]byte(test.responseBody))
+				assert.NoError(t, err)
+			}))
+			client := lidarr.New(starr.New("mockAPIkey", mockServer.URL, 0))
+			output, err := client.UpdateTag(test.tag)
+			assert.ErrorIs(t, err, test.withError, "error is not the same as expected")
+			assert.EqualValues(t, output, test.expectedResponse, "response is not the same as expected")
```

Comment:
`output` and `test.expectedResponse` are transposed. In all the files.

Reply:
good catch, I'll make a PR to fix this during this week!